python/singa/autograd.py

```diff
@@ -758,11 +731,83 @@ def __call__(self, x):
         else:
             if not hasattr(self, 'handle'):
                 self.handle = singa.CudnnConvHandle(x.data, self.kernel_size, self.stride,
-                                                    self.padding, self.in_channels, self.out_channels, self.bias)
+                                                    self.padding, self.in_channels, self.out_channels, self.bias,
+                                                    self.inner_params['workspace_MB_limit'] * 1024 * 1024, self.inner_params['cudnn_prefer'])
             elif x.shape[0] != self.handle.batchsize:
                 self.handle = singa.CudnnConvHandle(x.data, self.kernel_size, self.stride,
-                                                    self.padding, self.in_channels, self.out_channels, self.bias)
+                                                    self.padding, self.in_channels, self.out_channels, self.bias,
+                                                    self.inner_params['workspace_MB_limit'] * 1024 * 1024, self.inner_params['cudnn_prefer'])
+
         self.handle.device_id = x.device.id()
 
         y = conv2d(x, self.W, self.b, self.handle)
         return y
+
+
+class MaxPool2D(Layer):
```

Comment:
implement a Pool2D layer and then implment MaxPool2D and AvgPool2D?

Reply:
just need to change the argument form "MAX" to "AVERAGE" when initial handle,, then MaxPool2D layer changes to AvgPool2D layer